eZ/Publish/Core/MVC/Symfony/Routing/UrlWildcardRouter.php

```diff
@@ -0,0 +1,169 @@
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     *
+     * @return array An array of parameters
+     *
+     * @throws \Symfony\Component\Routing\Exception\ResourceNotFoundException
+     */
+    public function matchRequest(Request $request): array
+    {
+        $requestedPath = $request->attributes->get('semanticPathinfo', $request->getPathInfo());
+
+        try {
+            $urlWildcardTranslationResult = $this->wildcardService->translate($requestedPath);
+        } catch (NotFoundException $e) {
+            throw new ResourceNotFoundException($e->getMessage(), $e->getCode(), $e);
+        }
+
+        $this->logger->info("UrlWildcard matched. Destination URL: {$urlWildcardTranslationResult->uri}");
+
+        // redirect if URLWildcard is configured as forward
+        if ($urlWildcardTranslationResult->forward) {
+            $params = [
+                '_controller' => 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction',
+                'path' => $urlWildcardTranslationResult->uri,
+                'keepRequestMethod' => true,
+            ];
+
+            return $params;
+        }
+
+        // set translated path for the next router
+        $request->attributes->set('semanticPathinfo', $urlWildcardTranslationResult->uri);
+        $request->attributes->set('needsRedirect', false);
+
+        // and throw Exception to pass processing to the next router
+        throw new ResourceNotFoundException();
+    }
+
+    public function getRouteCollection(): RouteCollection
+    {
+        return new RouteCollection();
+    }
+
+    public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
+    {
+        throw new RouteNotFoundException('Could not match route');
+    }
+
+    public function setContext(RequestContext $context): void
+    {
+        $this->requestContext = $context;
+        $this->generator->setRequestContext($context);
+    }
+
+    public function getContext(): RequestContext
+    {
+        $this->requestContext;
+    }
+
+    /**
+     * Not supported. Please use matchRequest() instead.
+     *
+     * @param $pathinfo
+     *
+     * @return array
+     *
+     * @throws \RuntimeException
+     */
+    public function match($pathinfo): array
+    {
+        throw new \RuntimeException("The UrlWildcardRouter doesn't support the match() method. Please use matchRequest() instead.");
+    }
+
+    /**
+     * Whether the router supports the thing in $name to generate a route.
+     *
+     * This check does not need to look if the specific instance can be
+     * resolved to a route, only whether the router can generate routes from
+     * objects of this class.
+     *
+     * @param mixed $name The route name or route object
+     *
+     * @return bool
+     */
+    public function supports($name): bool
+    {
+        return $name === self::URL_ALIAS_ROUTE_NAME;
```

Comment:
If you want this entire feature to be enabled/disabled, then at the first glance this seems the right place to do it, instead of some container hacking via compiler passes :)
Still don't see a point though.

Reply:
Not really. This way, I would have to do an additional `if` which leads to (tiny, but still, please keep in mind that this piece of code will be executed on **every** request) additional load. 
I think it's better to not define the service at all if the feature it's not enabled.

Reply:
Defining service dynamically leads to a lot of magic difficult to track.
I'm really not following issue with additional overhead here.
Something like:
```php
return $isEnabled && $name === self::URL_ALIAS_ROUTE_NAME;
```
This way lookup done by this router is not executed unless feature is enabled.

Reply:
I found a better way to do so, done in: https://github.com/ezsystems/ezpublish-kernel/pull/2609/commits/5c923c1750b5c1973b64cacca5540a888045f35b